### PR TITLE
Netbox site module

### PR DESCRIPTION
**Pull request by FragmentedPacket**
##### SUMMARY

Adds a Netbox module to create/delete sites

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

netbox_site.py
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by smolz** [comment]
Tested and working for me on Ansible 2.8.0.dev0 and latest version of Netbox

File: lib/ansible/modules/net_tools/netbox/netbox_site.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+  - Creates or removes sites from Netbox
+notes:
+  - Tags should be defined as a YAML list
+  - This should be ran with connection C(local) and hosts C(localhost)
+author:
+  - Mikhail Yohman (@FragmentedPacket)
+requirements:
+  - pynetbox
+version_added: "2.8"
+options:
```
Comment (gundalow) [inline lib/ansible/modules/net_tools/netbox/netbox_site.py:28]: We are working on improving the module docs, could you please add `type: ` for each of the options, thanks.

File: lib/ansible/modules/net_tools/netbox/netbox_site.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+  netbox_url:
+    description:
+      - URL of the Netbox instance resolvable by Ansible control host
+    required: true
```
Comment (gundalow) [inline lib/ansible/modules/net_tools/netbox/netbox_site.py:32]:
```suggestion
    required: true
    type: str
```

File: lib/ansible/modules/net_tools/netbox/netbox_site.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+          - Retired
+      region:
+        description:
+          - The region that the site should be associated with
+      tenant:
+        description:
+          - The tenant the site will be assigned to
+      facility:
+        description:
+          - |
```
Comment (gundalow) [inline lib/ansible/modules/net_tools/netbox/netbox_site.py]:
```suggestion
          - Data center provider or facility.
          - eg C(Equinix NY7)
```

File: lib/ansible/modules/net_tools/netbox/netbox_site.py [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+            Data center provider or facility
+            ex. Equinix NY7
+      asn:
+        description:
+          - The ASN associated with the site
+        type: int
+      time_zone:
+        description:
+          - |
```
Comment (gundalow) [inline lib/ansible/modules/net_tools/netbox/netbox_site.py]: As above

File: lib/ansible/modules/net_tools/netbox/netbox_site.py [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+          - must exist in Netbox
+    required: true
+  state:
+    description:
+      - Use C(present) or C(absent) for adding or removing.
+    choices: [ absent, present ]
+    default: present
+  validate_certs:
+    description:
+      - |
```
Comment (gundalow) [inline lib/ansible/modules/net_tools/netbox/netbox_site.py:129]: Single line should be fine, we allow upto 160 chars

File: lib/ansible/modules/net_tools/netbox/netbox_site.py [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+        netbox_token=dict(type="str", required=True, no_log=True),
+        data=dict(type="dict", required=True),
+        state=dict(required=False, default="present", choices=["present", "absent"]),
+        validate_certs=dict(type="bool", default=True)
+    )
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=False)
+    # Fail module if pynetbox is not installed
+    if not HAS_PYNETBOX:
+        module.fail_json(msg=missing_required_lib('pynetbox'), exception=PYNETBOX_IMP_ERR)
```
Comment (gundalow) [inline lib/ansible/modules/net_tools/netbox/netbox_site.py:236]: Thanks for using `missing_required_lib`

File: lib/ansible/modules/net_tools/netbox/netbox_site.py (hunk 1 above)
Comment (FragmentedPacket) [reply lib/ansible/modules/net_tools/netbox/netbox_site.py:28]: Added type:

File: lib/ansible/modules/net_tools/netbox/netbox_site.py (hunk 2 above)
Comment (FragmentedPacket) [reply lib/ansible/modules/net_tools/netbox/netbox_site.py:32]: Done

**Comment by NilashishC** [comment]
@FragmentedPacket Could you please add some unit tests for this new module?

**Review by NilashishC — CHANGES_REQUESTED** [review]
@FragmentedPacket Could you please add some unit tests for this new module?

**Comment by FragmentedPacket** [comment]
@NilashishC I'm not sure unit tests will work in this case as most of the stuff returned here is an object itself so mocking that is pretty difficult. I was looking at integration testing. Is there a spot for more detailed documentation for building integration tests? We'd want to spin up a netbox instance using docker and then run the integration playbooks against that instance.

**Comment by FragmentedPacket** [comment]
@Anthony25 Do you have any ideas for testing strategies?

File: lib/ansible/modules/net_tools/netbox/netbox_site.py [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+    """
+    argument_spec = dict(
+        netbox_url=dict(type="str", required=True),
+        netbox_token=dict(type="str", required=True, no_log=True),
+        data=dict(type="dict", required=True),
+        state=dict(required=False, default="present", choices=["present", "absent"]),
+        validate_certs=dict(type="bool", default=True)
+    )
+
+    global module
```
Comment (NilashishC) [inline lib/ansible/modules/net_tools/netbox/netbox_site.py:230]: Please refrain from using `global`.

**Review by NilashishC — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/net_tools/netbox/netbox_site.py (hunk 7 above)
Comment (NilashishC) [reply lib/ansible/modules/net_tools/netbox/netbox_site.py:230]: As discussed in the other PR, please have this removed as a part of your refactoring.

**Review by NilashishC — APPROVED** [review]

**Review by gundalow — APPROVED** [review]

**Pull request merged**